File: frontend/src/api/writtenTest.ts
/**
 * 高校教学就业平台 - 笔试相关 API
 */
import request, { http } from '@/utils/request'

export interface WrittenTest {
  id?: number
  applicationId: number
  positionId: number
  studentId: number
  testUrl?: string
  duration?: number
  startTime?: string
  endTime?: string
  score?: number
  totalScore?: number
  status: 'pending' | 'ongoing' | 'completed' | 'missed'
  comment?: string
  isDeleted?: number
  createTime?: string
  updateTime?: string
  // 关联字段
  studentName?: string
  positionName?: string
  application?: any
  position?: any
  student?: any
}

export interface WrittenTestQueryParams {
  current?: number
  size?: number
  applicationId?: number
  positionId?: number
  studentId?: number
  status?: string
}

export const writtenTestApi = {
  /**
   * 分页查询笔试列表
   */
  getPage: (params: WrittenTestQueryParams) => {
    return http.get<any>('/written-test/page', { params })
  },

  /**
   * 根据ID获取笔试
   */
  getById: (id: number) => {
    return http.get<WrittenTest>(`/written-test/${id}`)
  },

  /**
   * 根据申请ID获取笔试信息
   */
  getByApplicationId: (applicationId: number) => {
    return http.get<WrittenTest>(`/written-test/application/${applicationId}`)
  },

  /**
   * 根据学生ID查询笔试列表
   */
  getByStudentId: (studentId: number, current = 1, size = 10) => {
    return http.get<any>(`/written-test/student/${studentId}`, { params: { current, size } })
  },

  /**
   * 新增笔试
   */
  create: (data: WrittenTest) => {
    return http.post<boolean>('/written-test', data)
  },

  /**
   * 更新笔试
   */
  update: (data: WrittenTest) => {
    return http.put<boolean>('/written-test', data)
  },

  /**
   * 删除笔试
   */
  delete: (id: number) => {
    return http.delete<boolean>(`/written-test/${id}`)
  },

  /**
   * 安排笔试
   */
  schedule: (data: {
    applicationId: number
    testUrl: string
    duration: number
    startTime: string
    endTime: string
  }) => {
    return http.post<boolean>('/written-test/schedule', null, { params: data })
  },

  /**
   * 学生开始考试
   */
  start: (id: number) => {
    return http.post<boolean>(`/written-test/${id}/start`)
  },

  /**
   * 提交笔试成绩
   */
  submitScore: (data: {
    id: number
    score: number
    totalScore: number
    comment?: string
  }) => {
    return http.post<boolean>(`/written-test/${data.id}/submit-score`, null, {
      params: {
        score: data.score,
        totalScore: data.totalScore,
        comment: data.comment
      }
    })
  },

  /**
   * 标记为缺席
   */
  markMissed: (id: number) => {
    return http.post<boolean>(`/written-test/${id}/mark-missed`)
  }
}

File: backend/src/main/java/com/teaching/employment/entity/WrittenTest.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 笔试实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-12
 */
@Data
@TableName("t_written_test")
@ApiModel(description = "笔试信息")
public class WrittenTest implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "笔试ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "申请ID")
    private Long applicationId;

    @ApiModelProperty(value = "职位ID")
    private Long positionId;

    @ApiModelProperty(value = "学生ID")
    private Long studentId;

    @ApiModelProperty(value = "考试链接或试卷URL")
    private String testUrl;

    @ApiModelProperty(value = "考试时长（分钟）")
    private Integer duration;

    @ApiModelProperty(value = "开始时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime startTime;

    @ApiModelProperty(value = "结束时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime endTime;

    @ApiModelProperty(value = "得分")
    private Integer score;

    @ApiModelProperty(value = "总分")
    private Integer totalScore;

    @ApiModelProperty(value = "考试状态：pending-待考试 ongoing-考试中 completed-已完成 missed-缺席")
    private String status;

    @ApiModelProperty(value = "评语")
    private String comment;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 申请信息（关联查询时使用，不映射到数据库字段）
     */
    @TableField(exist = false)
    private JobApplication application;

    /**
     * 职位信息
     */
    @TableField(exist = false)
    private Position position;

    /**
     * 学生信息
     */
    @TableField(exist = false)
    private Student student;

    /**
     * 学生姓名
     */
    @TableField(exist = false)
    private String studentName;

    /**
     * 职位名称
     */
    @TableField(exist = false)
    private String positionName;
}
